plugins/modules/azure_rm_apimanagement.py

```diff
@@ -0,0 +1,631 @@
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.body[key] = kwargs[key]
+
+        # https://docs.microsoft.com/en-us/azure/templates/microsoft.apimanagement/service/apis
+        self.inflate_parameters(self.module_arg_spec, self.body, 0)
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(ApiManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager,
+                                                    api_version='2020-06-01-preview')
+
+        old_response = self.get_resource()
+
+        if old_response is False:
+            self.log("Api instance does not exist in the given service.")
+            if self.state == 'present':
+                self.to_do = Actions.Create
+            else:
+                self.log("Old instance didn't exist")
+        else:
+            self.log("Api instance already exists in the given service.")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            else:
+                modifiers = {}
+                self.create_compare_modifiers(self.module_arg_spec, '', modifiers)
+                self.results['modifiers'] = modifiers
+                self.results['compare'] = []
+                if not self.default_compare(modifiers, self.body, old_response, '', self.results):
+                    self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log('Create and Update the Api instance.')
+
+            if self.check_mode:
+                self.results['changed'] = True
+                return self.results
+
+            response = self.create_and_update_resource()
+            self.results['changed'] = True
+
+        elif self.to_do == Actions.Delete:
+            self.log('Api instance deleted.')
+
+            if self.check_mode:
+                self.results['changed'] = True
+                return self.results
+
+            self.delete_resource()
+            self.results['changed'] = True
+        else:
+            self.log('No change in Api instance.')
+            self.results['changed'] = False
+            response = old_response
+
+        return self.results
+
+    # This function will create and update resource on the api management service.
+    def create_and_update_resource(self):
+
+        try:
+          response = self.mgmt_client.api.create_or_update(resource_group_name=self.resource_group,
+                                                           service_name=self.service_name,
+                                                           api_id=self.api_id,
+                                                           parameters=self.body,
+							   if_match="*")
```

Comment:
indentation contains mixed spaces and tabs and  continuation line under-indented for visual indent